std/json.d

```diff
@@ -213,13 +213,15 @@ struct JSONValue
      * Value getter/setter for `JSONType.float_`. Note that despite
      * the name, this is a $(B 64)-bit `double`, not a 32-bit `float`.
      * Throws: `JSONException` for read access if `type` is not
-     * `JSONType.float_`.
+     * `JSONType.float_`, `JSONType.integer` or `JSONType.uinteger`.
      */
     @property double floating() const pure @safe
     {
-        enforce!JSONException(type == JSONType.float_,
-                                "JSONValue is not a floating type");
-        return store.floating;
+        if(type == JSONType.float_) return store.floating;
+        if(type == JSONType.uinteger) return cast(double)store.uinteger;
+        if(type == JSONType.integer) return cast(double)store.integer;
+
+        throw new JSONException("JSONValue is not a floating type");
```

Comment:
This is not covered by the tests. You can use `assertThrown` from `std.exception` for that purpose.

Reply:
I will add a test case for this exception, thank you!